hack/local-up-cluster.sh

```diff
@@ -509,11 +510,23 @@ EOF
 # start_discovery relies on certificates created by start_apiserver
 function start_discovery {
     # TODO generate serving certificates
-    
+    create_client_certkey client-ca discovery-auth system:discovery-auth
+    write_client_kubeconfig discovery-auth
+
+    # grant permission to run delegated authentication and authorization checks
+    kubectl create clusterrolebinding discovery:system:auth-delegator --clusterrole=system:auth-delegator --user=system:discovery-auth
+
     DISCOVERY_SERVER_LOG=/tmp/kubernetes-discovery.log
     ${CONTROLPLANE_SUDO} "${GO_OUT}/kubernetes-discovery" \
       --cert-dir="${CERT_DIR}" \
       --client-ca-file="${CERT_DIR}/client-ca-bundle.crt" \
+      --authentication-kubeconfig="${CERT_DIR}/discovery-auth.kubeconfig" \
```

Comment:
> is this the discovery server accepting requestheader auth, or knowing what headers to use to send it along to the underlying servers? I expected the apiserver to add these in order to trust the proxied requests from the discovery server

This flag is used to check tokenreviews.  The request header flags are used to accept an auth-proxy.  The process will *also* attach proxy headers on proxied requests, but that isn't wired up yet.

The core API server already has the request-header flags in local-up-cluster.sh